src/main/java/school/lemon/changerequest/java/io/hw1/IFileManagerImp.java

```diff
@@ -0,0 +1,343 @@
+                    Arrays.sort(paths, comparatorForLexDescendingNoOrder);
+                    for (File file : paths) {
+                        result.add(file);
+                    }
+                    break;
+                case BY_LAST_UPDATE_DESCENDING:
+                    Arrays.sort(paths, comparatorForLastUpdateDescendingNoOrder);
+                    for (File file : paths) {
+                        result.add(file);
+                    }
+                    break;
+                case BY_LAST_UPDATE_ASCENDING:
+                    Arrays.sort(paths, comparatorForLastUpdateAscendingNoOrder);
+                    for (File file : paths) {
+                        result.add(file);
+                    }
+                    break;
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public String fileInfo(File file) {
+        if (!file.exists()) {
+            return null;
+        }
+        long size;
+        String fileName;
+        if (file.isDirectory()) {
+            size = 0;
+        } else {
+            size = file.length();
+        }
+        SimpleDateFormat sdf = new SimpleDateFormat("dd.MM.yyyy HH:mm:ss");
+        String lastModifiedDate = sdf.format(file.lastModified());
+        if (file.isDirectory()) fileName = file.getName() + File.separator;
+        else fileName = file.getName();
+        return String.format("%-20s|%15s|%s %n", lastModifiedDate, size, fileName);
+    }
+
+    @Override
+    public void createFile(String name) throws FileAlreadyExistException, FileManagerException {
+        File file = new File(name);
+        if (file.exists()) throw new FileAlreadyExistException(file);
+        try {
+            file.createNewFile();
+        } catch (IOException e) {
+        }
+        if (!file.exists()) throw new FileManagerException();
+    }
+
+
+    @Override
+    public void createDirectory(String name) throws DirectoryAlreadyExistException, FileManagerException {
+        File file = new File(name);
+        if (file.exists()) throw new DirectoryAlreadyExistException(file);
+        file.mkdir();
+        if (!file.exists()) throw new FileManagerException();
+    }
+
+    @Override
+    public void changeDirectory(String name) throws IllegalDirectoryException {
+        File file = new File(name);
+        if (!file.isDirectory()) {
+            throw new IllegalDirectoryException(file);
+        }
+        setCurrentDirectory(name);
+    }
+
+    @Override
+    public boolean delete(String name) throws FileNotFoundException {
+        File file = new File(name);
+        if (!file.exists()) {
+            throw new FileNotFoundException();
+        }
+        if (file.isDirectory()) {
+            if (file.list().length > 0) return false;
+        }
+        file.delete();
+        return true;
+
+    }
+
+
+    @Override
+    public boolean delete(String name, boolean force) throws FileNotFoundException {
+        File[] files = currentDirectory.listFiles();
```

Comment:
your implementation for overloaded methods should follow opposite principle: you must call wider method from thinner.